src/controller.rs

```diff
@@ -45,7 +45,10 @@ impl<'b> Controller<'b> {
             match input_file.get_reader(&stdin) {
                 Err(error) => {
                     handle_error(&error);
-                    no_errors = false;
+                    match error {
```

Comment:
Instead of matching on `ErrorKind::ImmediateEOF` here and in `handle_error`, could we also move the `handle_error` call into the `_ => ` branch?

``` rust
match error {
    Error(ErrorKind::ImmediateEOF, _) => (),
    _ => {
        handle_error(&error);
        no_errors = false
    },
}
```